Project.Xml.Test/XmlTest.cs

```diff
@@ -0,0 +1,203 @@
+
+            //act
+            //create the testresults sorted with nulls at the end
+            string[] testResults = Program.GetSortedData(ifhXml.GetParsedData(xml)).ToArray();
+
+            //assert
+            //validate that the sorted testresults match the sorted expected results
+            for (int i = 0; i < expectedResults.Length; i++)
+            {
+                Assert.AreEqual(testResults[i], expectedResults[i]);
+            }
+        }
+
+        [TestMethod]
+        public void File_WithValidXMLNumericDataSorted()
+        {
+            //arrange
+            IFileHandling ifhXml = new XmlParser();
+
+            //create xml string
+            string xml = "<xml> <age id = '55' /><age id = '92' /><age id = '25' /><orange /></xml>";
+
+            //create the expected results in the expected sort order
+            string[] expectedResults = new string[] { "25", "55", "92", null };
+
+            //act
+            //create the testresults sorted with nulls at the end
+            string[] testResults = Program.GetSortedData(ifhXml.GetParsedData(xml)).ToArray();
+
+            //assert
+            //validate that the sorted testresults match the sorted expected results
+            for (int i = 0; i < expectedResults.Length; i++)
+            {
+                Assert.AreEqual(testResults[i], expectedResults[i]);
+            }
+        }
+
+        [TestMethod]
+        public void File_WithValidXMLAlphaNumericDataSorted()
+        {
+            //arrange
+            IFileHandling ifhXml = new XmlParser();
+
+            //create xml string
+            string xml = "<xml> <orange id = 'Round Orange' /><orange id = 'Naval Orange' /><orange /><age id = '55' /><age id = '92' /><age /></xml>";
+
+            //create the expected results in the expected sort order
+            string[] expectedResults = new string[] { "55", "92", "Naval Orange", "Round Orange", null, null };
+
+            //act
+            //create the testresults sorted with nulls at the end
+            string[] testResults = Program.GetSortedData(ifhXml.GetParsedData(xml)).ToArray();
+
+            //assert
+            //validate that the sorted testresults match the sorted expected results
+            for (int i = 0; i < expectedResults.Length; i++)
+            {
+                Assert.AreEqual(testResults[i], expectedResults[i]);
+            }
+        }
+
+        #endregion
+
+        #region  " Non Test Methods "
+
+        /// <summary>
+        /// CreateTempFile
+        /// </summary>
+        /// <param name="fileData">array containing data you want to write to the file. pass an empty array if no data needed</param>
+        /// <returns>filepath of new temp file</returns>
+        string CreateTempFile(string[] fileData)
+        {
+            string file = Path.GetTempFileName();
+
+            if (fileData.Length > 0)
```

Comment:
`Observation`: This can simply be written as `File.WriteAllLines(file, fileData);`.